ruby/ql/lib/codeql/ruby/security/XpathInjectionQuery.qll

```diff
@@ -0,0 +1,24 @@
+/**
+ * Provides a taint-tracking configuration for detecting "Xpath Injection" vulnerabilities.
+ *
+ * Note, for performance reasons: only import this file if
+ * `XpathInjection::Configuration` is needed, otherwise
+ * `XpathInjectionCustomizations` should be imported instead.
+ */
+
+private import codeql.ruby.DataFlow
+private import codeql.ruby.TaintTracking
+import XpathInjectionCustomizations::XpathInjection
+
+/**
+ * A taint-tracking configuration for detecting "Xpath Injection" vulnerabilities.
+ */
+class Configuration extends TaintTracking::Configuration {
+  Configuration() { this = "Xpath Injection" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+  override predicate isSanitizer(DataFlow::Node node) { node instanceof Sanitizer }
+}
```

Comment:
We've recently changed how dataflow configurations are written to use a module implementing `DataFlow::ConfigSig` instead of a class extending `TaintTracking::Configuration`. There's an example of this in https://github.com/github/codeql/pull/12419/files - would you mind making this query use the new format instead, as this will save having to deprecate and port this configuration later on?